### ISSUE TITLE
Predictive Search Function

### ISSUE THREAD

**Issue by kowalczj** [kowalczj/shop24#7]
As a customer, I want to be able to search and filter for the type of item I am looking for so that I can find the item I'm looking for faster. It would also be helpful if it can return search results as I type, so that I don't have to type the entire word in the search box.

**Comment by OSSAVIVI** [comment]
Jon and I will create the UI for the search bar and will connect to DB later on when

**Comment by crisischris** [comment]
(estimation) 1 week.  this is a tri tree auto fill implementation.  We would have to scan all items to build the tree and implement it in the UI.

**Comment by kraynakr** [comment]
Hope we're not building this from scratch, there are free frameworks out there like jquery auto complete widget: https://jqueryui.com/autocomplete/

**Comment by OSSAVIVI** [comment]
> Hope we're not building this from scratch, there are free frameworks out there like jquery auto complete widget: https://jqueryui.com/autocomplete/

Yes I was planning on using something pre-built from online (thx for sharing link). I chose this story because it seemed like an easy one to at least get the UI set up while we work out the kinks in the DB
